docs/book/src/migration/guide.md

```diff
@@ -0,0 +1,192 @@
+# Migration Guide
+
+This document describes how to migrate a project from kubebuilder v1 to v2.
+
+We suggest you read [this doc](./v1vsv2.md) that compares the notable differences between v1 and v2.
```

Comment:
Lang: Change to a more imperative sentence structure: `Make you understand the [differences between KubeBuilder v1 and v2](...)  before continuing`